[Bug]: ERROR - liquibase.changelog.ChangeSet  +  liquibase.exception.DatabaseException after latest Update

### What happened?

After latest Update container cant  start and have ERROR :  

![image](https://github.com/kagemomiji/airsonic-advanced/assets/28630321/4e9867c5-d528-4f2c-a4ef-46e59899e9d3)
![image](https://github.com/kagemomiji/airsonic-advanced/assets/28630321/ad901303-5972-416e-ac3d-c20ed2758872)






### Steps to reproduce

update from privies version

### Version

11.1.x (Edge)

### Version Detail

latest

### Operating System

Docker in Unraid (linux x64)

### Java Version

java 17.0.5

### Database

MariaDB 8.x

### DB Detail

_No response_

### Configuration paramter

...

### Proxy Server

_No response_

### client detail

_No response_

### language

None

### Relevant log output

```console
04.12.2023
00:58:33
04.12.2023
00:58:33
2023-12-04 00:58:33.242  INFO --- org.airsonic.player.Application          : Starting Application using Java 17.0.5 on a91bd0388166 with PID 1 (/app/WEB-INF/classes started by au in /var)
04.12.2023
00:58:33
2023-12-04 00:58:33.244  INFO --- org.airsonic.player.Application          : No active profile set, falling back to 1 default profile: "default"
04.12.2023
00:58:35
2023-12-04 00:58:35.743  INFO --- o.a.p.service.SettingsService            : Java: 17.0.5+8, OS: Linux, Memory (max bytes): 268435456
04.12.2023
00:58:35
2023-12-04 00:58:35.849  INFO --- org.airsonic.player.Application          : Detected Tomcat web server
04.12.2023
00:58:37
2023-12-04 00:58:37.969  INFO --- l.l.StandardLockService                  : Successfully acquired change log lock
04.12.2023
00:58:39
2023-12-04 00:58:39.243  INFO --- liquibase.Liquibase                      : Parsed changelog file 'classpath:liquibase/db-changelog.xml'
04.12.2023
00:58:39
2023-12-04 00:58:39.284  INFO --- l.c.StandardChangeLogHistoryService      : Reading from `airsonic-advanced`.DATABASECHANGELOG
04.12.2023
00:58:39
2023-12-04 00:58:39.519  INFO --- l.l.StandardLockService                  : Successfully released change log lock
04.12.2023
00:58:39
2023-12-04 00:58:39.587  INFO --- l.l.StandardLockService                  : Successfully acquired change log lock
04.12.2023
00:58:40
2023-12-04 00:58:40.479  INFO --- l.c.StandardChangeLogHistoryService      : Reading from `airsonic-advanced`.DATABASECHANGELOG
04.12.2023
00:58:40
2023-12-04 00:58:40.587  INFO --- l.command.core.UpdateCommandStep         : Using deploymentId: 1644320488
04.12.2023
00:58:40
2023-12-04 00:58:40.616  INFO --- l.c.StandardChangeLogHistoryService      : Reading from `airsonic-advanced`.DATABASECHANGELOG
04.12.2023
00:58:40
Running Changeset: liquibase/11.1/add-genre-pk.xml::add_genre_pk::kagemomiji
04.12.2023
00:58:40
2023-12-04 00:58:40.712  WARN --- o.m.j.message.server.ErrorPacket         : Error: 1062-23000: Duplicate entry 'AudioBook' for key 'PRIMARY'
04.12.2023
00:58:40
2023-12-04 00:58:40.714 ERROR --- liquibase.changelog.ChangeSet            : ChangeSet liquibase/11.1/add-genre-pk.xml::add_genre_pk::kagemomiji encountered an exception.
04.12.2023
00:58:40
2023-12-04 00:58:40.716  INFO --- l.command.core.UpdateCommandStep         : Update command encountered an exception.
04.12.2023
00:58:40
2023-12-04 00:58:40.719  INFO --- l.l.StandardLockService                  : Successfully released change log lock
04.12.2023
00:58:40
2023-12-04 00:58:40.721  INFO --- l.l.StandardLockService                  : Successfully released change log lock
04.12.2023
00:58:40
2023-12-04 00:58:40.721  INFO --- liquibase.command.CommandScope           : Command execution complete
04.12.2023
00:58:40
2023-12-04 00:58:40.727  WARN --- ConfigServletWebServerApplicationContext : Exception encountered during context initialization - cancelling refresh attempt: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'liquibase' defined in class path resource [org/springframework/boot/autoconfigure/liquibase/LiquibaseAutoConfiguration$LiquibaseConfiguration.class]: Invocation of init method failed; nested exception is liquibase.exception.CommandExecutionException: liquibase.exception.LiquibaseException: liquibase.exception.MigrationFailedException: Migration failed for changeset liquibase/11.1/add-genre-pk.xml::add_genre_pk::kagemomiji:
04.12.2023
00:58:40
     Reason: liquibase.exception.DatabaseException: (conn=5562) Duplicate entry 'AudioBook' for key 'PRIMARY' [Failed SQL: (1062) ALTER TABLE `airsonic-advanced`.genre ADD PRIMARY KEY (name)]
04.12.2023
00:58:40
2023-12-04 00:58:40.801 ERROR --- o.s.boot.SpringApplication               : Application run failed
04.12.2023
00:58:40
04.12.2023
00:58:40
org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'liquibase' defined in class path resource [org/springframework/boot/autoconfigure/liquibase/LiquibaseAutoConfiguration$LiquibaseConfiguration.class]: Invocation of init method failed; nested exception is liquibase.exception.CommandExecutionException: liquibase.exception.LiquibaseException: liquibase.exception.MigrationFailedException: Migration failed for changeset liquibase/11.1/add-genre-pk.xml::add_genre_pk::kagemomiji:
04.12.2023
00:58:40
     Reason: liquibase.exception.DatabaseException: (conn=5562) Duplicate entry 'AudioBook' for key 'PRIMARY' [Failed SQL: (1062) ALTER TABLE `airsonic-advanced`.genre ADD PRIMARY KEY (name)]
04.12.2023
00:58:40
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1804) ~[spring-beans-5.3.30.jar:5.3.30]
04.12.2023
00:58:40
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:620) ~[spring-beans-5.3.30.jar:5.3.30]
04.12.2023
00:58:40
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:542) ~[spring-beans-5.3.30.jar:5.3.30]
04.12.2023
00:58:40
	at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:335) ~[spring-beans-5.3.30.jar:5.3.30]
04.12.2023
00:58:40
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:234) ~[spring-beans-5.3.30.jar:5.3.30]
04.12.2023
00:58:40
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:333) ~[spring-beans-5.3.30.jar:5.3.30]
04.12.2023
00:58:40
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:208) ~[spring-beans-5.3.30.jar:5.3.30]
04.12.2023
00:58:40
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:322) ~[spring-beans-5.3.30.jar:5.3.30]
04.12.2023
00:58:40
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:208) ~[spring-beans-5.3.30.jar:5.3.30]
04.12.2023
00:58:40
	at org.springframework.context.support.AbstractApplicationContext.getBean(AbstractApplicationContext.java:1160) ~[spring-context-5.3.30.jar:5.3.30]
04.12.2023
00:58:40
	at org.springframework.context.support.AbstractApplicationContext.finishBeanFactoryInitialization(AbstractApplicationContext.java:911) ~[spring-context-5.3.30.jar:5.3.30]
04.12.2023
00:58:40
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:583) ~[spring-context-5.3.30.jar:5.3.30]
04.12.2023
00:58:40
	at org.springframework.boot.web.servlet.context.ServletWebServerApplicationContext.refresh(ServletWebServerApplicationContext.java:147) ~[spring-boot-2.7.16.jar:2.7.16]
04.12.2023
00:58:40
	at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:731) ~[spring-boot-2.7.16.jar:2.7.16]
04.12.2023
00:58:40
	at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:408) ~[spring-boot-2.7.16.jar:2.7.16]
04.12.2023
00:58:40
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:307) ~[spring-boot-2.7.16.jar:2.7.16]
04.12.2023
00:58:40
	at org.springframework.boot.builder.SpringApplicationBuilder.run(SpringApplicationBuilder.java:164) ~[spring-boot-2.7.16.jar:2.7.16]
04.12.2023
00:58:40
	at org.airsonic.player.Application.main(Application.java:70) ~[classes/:na]
04.12.2023
00:58:40
Caused by: liquibase.exception.CommandExecutionException: liquibase.exception.LiquibaseException: liquibase.exception.MigrationFailedException: Migration failed for changeset liquibase/11.1/add-genre-pk.xml::add_genre_pk::kagemomiji:
04.12.2023
00:58:40
     Reason: liquibase.exception.DatabaseException: (conn=5562) Duplicate entry 'AudioBook' for key 'PRIMARY' [Failed SQL: (1062) ALTER TABLE `airsonic-advanced`.genre ADD PRIMARY KEY (name)]
04.12.2023
00:58:40
	at liquibase.command.CommandScope.execute(CommandScope.java:235) ~[liquibase-core-4.22.0.jar:na]
04.12.2023
00:58:40
	at liquibase.Liquibase.lambda$update$0(Liquibase.java:237) ~[liquibase-core-4.22.0.jar:na]
04.12.2023
00:58:40
	at liquibase.Scope.lambda$child$0(Scope.java:194) ~[liquibase-core-4.22.0.jar:na]
04.12.2023
00:58:40
	at liquibase.Scope.child(Scope.java:203) ~[liquibase-core-4.22.0.jar:na]
04.12.2023
00:58:40
	at liquibase.Scope.child(Scope.java:193) ~[liquibase-core-4.22.0.jar:na]
04.12.2023
00:58:40
	at liquibase.Scope.child(Scope.java:172) ~[liquibase-core-4.22.0.jar:na]
04.12.2023
00:58:40
	at liquibase.Liquibase.runInScope(Liquibase.java:1605) ~[liquibase-core-4.22.0.jar:na]
04.12.2023
00:58:40
	at liquibase.Liquibase.update(Liquibase.java:229) ~[liquibase-core-4.22.0.jar:na]
04.12.2023
00:58:40
	at liquibase.Liquibase.update(Liquibase.java:213) ~[liquibase-core-4.22.0.jar:na]
04.12.2023
00:58:40
	at liquibase.integration.spring.SpringLiquibase.performUpdate(SpringLiquibase.java:329) ~[liquibase-core-4.22.0.jar:na]
04.12.2023
00:58:40
	at liquibase.integration.spring.SpringLiquibase.afterPropertiesSet(SpringLiquibase.java:284) ~[liquibase-core-4.22.0.jar:na]
04.12.2023
00:58:40
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.invokeInitMethods(AbstractAutowireCapableBeanFactory.java:1863) ~[spring-beans-5.3.30.jar:5.3.30]
04.12.2023
00:58:40
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1800) ~[spring-beans-5.3.30.jar:5.3.30]
04.12.2023
00:58:40
	... 17 common frames omitted
04.12.2023
00:58:40
Caused by: liquibase.exception.LiquibaseException: liquibase.exception.MigrationFailedException: Migration failed for changeset liquibase/11.1/add-genre-pk.xml::add_genre_pk::kagemomiji:
04.12.2023
00:58:40
     Reason: liquibase.exception.DatabaseException: (conn=5562) Duplicate entry 'AudioBook' for key 'PRIMARY' [Failed SQL: (1062) ALTER TABLE `airsonic-advanced`.genre ADD PRIMARY KEY (name)]
04.12.2023
00:58:40
	at liquibase.changelog.ChangeLogIterator.run(ChangeLogIterator.java:121) ~[liquibase-core-4.22.0.jar:na]
04.12.2023
00:58:40
	at liquibase.command.core.AbstractUpdateCommandStep.lambda$run$0(AbstractUpdateCommandStep.java:103) ~[liquibase-core-4.22.0.jar:na]
04.12.2023
00:58:40
	at liquibase.Scope.lambda$child$0(Scope.java:194) ~[liquibase-core-4.22.0.jar:na]
04.12.2023
00:58:40
	at liquibase.Scope.child(Scope.java:203) ~[liquibase-core-4.22.0.jar:na]
04.12.2023
00:58:40
	at liquibase.Scope.child(Scope.java:193) ~[liquibase-core-4.22.0.jar:na]
04.12.2023
00:58:40
	at liquibase.Scope.child(Scope.java:172) ~[liquibase-core-4.22.0.jar:na]
04.12.2023
00:58:40
	at liquibase.command.core.AbstractUpdateCommandStep.run(AbstractUpdateCommandStep.java:100) ~[liquibase-core-4.22.0.jar:na]
04.12.2023
00:58:40
	at liquibase.command.CommandScope.execute(CommandScope.java:212) ~[liquibase-core-4.22.0.jar:na]
04.12.2023
00:58:40
	... 29 common frames omitted
04.12.2023
00:58:40
Caused by: liquibase.exception.MigrationFailedException: Migration failed for changeset liquibase/11.1/add-genre-pk.xml::add_genre_pk::kagemomiji:
04.12.2023
00:58:40
     Reason: liquibase.exception.DatabaseException: (conn=5562) Duplicate entry 'AudioBook' for key 'PRIMARY' [Failed SQL: (1062) ALTER TABLE `airsonic-advanced`.genre ADD PRIMARY KEY (name)]
04.12.2023
00:58:40
	at liquibase.changelog.ChangeSet.execute(ChangeSet.java:759) ~[liquibase-core-4.22.0.jar:na]
04.12.2023
00:58:40
	at liquibase.changelog.visitor.UpdateVisitor.visit(UpdateVisitor.java:61) ~[liquibase-core-4.22.0.jar:na]
04.12.2023
00:58:40
	at liquibase.changelog.ChangeLogIterator$2.lambda$null$0(ChangeLogIterator.java:109) ~[liquibase-core-4.22.0.jar:na]
04.12.2023
00:58:40
	at liquibase.Scope.lambda$child$0(Scope.java:194) ~[liquibase-core-4.22.0.jar:na]
04.12.2023
00:58:40
	at liquibase.Scope.child(Scope.java:203) ~[liquibase-core-4.22.0.jar:na]
04.12.2023
00:58:40
	at liquibase.Scope.child(Scope.java:193) ~[liquibase-core-4.22.0.jar:na]
04.12.2023
00:58:40
	at liquibase.Scope.child(Scope.java:172) ~[liquibase-core-4.22.0.jar:na]
04.12.2023
00:58:40
	at liquibase.changelog.ChangeLogIterator$2.lambda$run$1(ChangeLogIterator.java:109) ~[liquibase-core-4.22.0.jar:na]
04.12.2023
00:58:40
	at liquibase.Scope.lambda$child$0(Scope.java:194) ~[liquibase-core-4.22.0.jar:na]
04.12.2023
00:58:40
	at liquibase.Scope.child(Scope.java:203) ~[liquibase-core-4.22.0.jar:na]
04.12.2023
00:58:40
	at liquibase.Scope.child(Scope.java:193) ~[liquibase-core-4.22.0.jar:na]
04.12.2023
00:58:40
	at liquibase.Scope.child(Scope.java:172) ~[liquibase-core-4.22.0.jar:na]
04.12.2023
00:58:40
	at liquibase.Scope.child(Scope.java:260) ~[liquibase-core-4.22.0.jar:na]
04.12.2023
00:58:40
	at liquibase.changelog.ChangeLogIterator$2.run(ChangeLogIterator.java:92) ~[liquibase-core-4.22.0.jar:na]
04.12.2023
00:58:40
	at liquibase.Scope.lambda$child$0(Scope.java:194) ~[liquibase-core-4.22.0.jar:na]
04.12.2023
00:58:40
	at liquibase.Scope.child(Scope.java:203) ~[liquibase-core-4.22.0.jar:na]
04.12.2023
00:58:40
	at liquibase.Scope.child(Scope.java:193) ~[liquibase-core-4.22.0.jar:na]
04.12.2023
00:58:40
	at liquibase.Scope.child(Scope.java:172) ~[liquibase-core-4.22.0.jar:na]
04.12.2023
00:58:40
	at liquibase.Scope.child(Scope.java:260) ~[liquibase-core-4.22.0.jar:na]
04.12.2023
00:58:40
	at liquibase.Scope.child(Scope.java:264) ~[liquibase-core-4.22.0.jar:na]
04.12.2023
00:58:40
	at liquibase.changelog.ChangeLogIterator.run(ChangeLogIterator.java:66) ~[liquibase-core-4.22.0.jar:na]
04.12.2023
00:58:40
	... 36 common frames omitted
04.12.2023
00:58:40
Caused by: liquibase.exception.DatabaseException: (conn=5562) Duplicate entry 'AudioBook' for key 'PRIMARY' [Failed SQL: (1062) ALTER TABLE `airsonic-advanced`.genre ADD PRIMARY KEY (name)]
04.12.2023
00:58:40
	at liquibase.executor.jvm.JdbcExecutor$ExecuteStatementCallback.doInStatement(JdbcExecutor.java:445) ~[liquibase-core-4.22.0.jar:na]
04.12.2023
00:58:40
	at liquibase.executor.jvm.JdbcExecutor.execute(JdbcExecutor.java:77) ~[liquibase-core-4.22.0.jar:na]
04.12.2023
00:58:40
	at liquibase.executor.jvm.JdbcExecutor.execute(JdbcExecutor.java:160) ~[liquibase-core-4.22.0.jar:na]
04.12.2023
00:58:40
	at liquibase.database.AbstractJdbcDatabase.execute(AbstractJdbcDatabase.java:1287) ~[liquibase-core-4.22.0.jar:na]
04.12.2023
00:58:40
	at liquibase.database.AbstractJdbcDatabase.executeStatements(AbstractJdbcDatabase.java:1269) ~[liquibase-core-4.22.0.jar:na]
04.12.2023
00:58:40
	at liquibase.changelog.ChangeSet.execute(ChangeSet.java:717) ~[liquibase-core-4.22.0.jar:na]
04.12.2023
00:58:40
	... 56 common frames omitted
04.12.2023
00:58:40
Caused by: java.sql.SQLIntegrityConstraintViolationException: (conn=5562) Duplicate entry 'AudioBook' for key 'PRIMARY'
04.12.2023
00:58:40
	at org.mariadb.jdbc.export.ExceptionFactory.createException(ExceptionFactory.java:290) ~[mariadb-java-client-3.1.4.jar:na]
04.12.2023
00:58:40
	at org.mariadb.jdbc.export.ExceptionFactory.create(ExceptionFactory.java:370) ~[mariadb-java-client-3.1.4.jar:na]
04.12.2023
00:58:40
	at org.mariadb.jdbc.message.ClientMessage.readPacket(ClientMessage.java:134) ~[mariadb-java-client-3.1.4.jar:na]
04.12.2023
00:58:40
	at org.mariadb.jdbc.client.impl.StandardClient.readPacket(StandardClient.java:872) ~[mariadb-java-client-3.1.4.jar:na]
04.12.2023
00:58:40
	at org.mariadb.jdbc.client.impl.StandardClient.readResults(StandardClient.java:811) ~[mariadb-java-client-3.1.4.jar:na]
04.12.2023
00:58:40
	at org.mariadb.jdbc.client.impl.StandardClient.readResponse(StandardClient.java:730) ~[mariadb-java-client-3.1.4.jar:na]
04.12.2023
00:58:40
	at org.mariadb.jdbc.client.impl.StandardClient.execute(StandardClient.java:654) ~[mariadb-java-client-3.1.4.jar:na]
04.12.2023
00:58:40
	at org.mariadb.jdbc.Statement.executeInternal(Statement.java:957) ~[mariadb-java-client-3.1.4.jar:na]
04.12.2023
00:58:40
	at org.mariadb.jdbc.Statement.execute(Statement.java:1083) ~[mariadb-java-client-3.1.4.jar:na]
04.12.2023
00:58:40
	at org.mariadb.jdbc.Statement.execute(Statement.java:474) ~[mariadb-java-client-3.1.4.jar:na]
04.12.2023
00:58:40
	at com.zaxxer.hikari.pool.ProxyStatement.execute(ProxyStatement.java:94) ~[HikariCP-4.0.3.jar:na]
04.12.2023
00:58:40
	at com.zaxxer.hikari.pool.HikariProxyStatement.execute(HikariProxyStatement.java) ~[HikariCP-4.0.3.jar:na]
04.12.2023
00:58:40
	at liquibase.executor.jvm.JdbcExecutor$ExecuteStatementCallback.doInStatement(JdbcExecutor.java:441) ~[liquibase-core-4.22.0.jar:na]
04.12.2023
00:58:40
	... 61 common frames omitted
```

04.12.2023
08:50:21
2023-12-04 08:50:21.468  INFO --- l.c.StandardChangeLogHistoryService      : Reading from `airsonic-advanced`.DATABASECHANGELOG
04.12.2023
08:50:21
Running Changeset: liquibase/11.1/add-genre-pk.xml::add_genre_pk::kagemomiji
04.12.2023
08:50:21
2023-12-04 08:50:21.565  WARN --- o.m.j.message.server.ErrorPacket         : Error: 1062-23000: Duplicate entry 'AudioBook' for key 'PRIMARY'
04.12.2023
08:50:21
2023-12-04 08:50:21.568 ERROR --- liquibase.changelog.ChangeSet            : ChangeSet liquibase/11.1/add-genre-pk.xml::add_genre_pk::kagemomiji encountered an exception.
04.12.2023
08:50:21
2023-12-04 08:50:21.569  INFO --- l.command.core.UpdateCommandStep         : Update command encountered an exception.
04.12.2023
08:50:21
2023-12-04 08:50:21.572  INFO --- l.l.StandardLockService                  : Successfully released change log lock
04.12.2023
08:50:21
2023-12-04 08:50:21.574  INFO --- l.l.StandardLockService                  : Successfully released change log lock
04.12.2023
08:50:21
2023-12-04 08:50:21.574  INFO --- liquibase.command.CommandScope           : Command execution complete
04.12.2023
08:50:21
2023-12-04 08:50:21.577  WARN --- ConfigServletWebServerApplicationContext : Exception encountered during context initialization - cancelling refresh attempt: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'liquibase' defined in class path resource [org/springframework/boot/autoconfigure/liquibase/LiquibaseAutoConfiguration$LiquibaseConfiguration.class]: Invocation of init method failed; nested exception is liquibase.exception.CommandExecutionException: liquibase.exception.LiquibaseException: liquibase.exception.MigrationFailedException: Migration failed for changeset liquibase/11.1/add-genre-pk.xml::add_genre_pk::kagemomiji:
04.12.2023
08:50:21
     Reason: liquibase.exception.DatabaseException: (conn=254) Duplicate entry 'AudioBook' for key 'PRIMARY' [Failed SQL: (1062) ALTER TABLE `airsonic-advanced`.genre ADD PRIMARY KEY (name)]
04.12.2023
08:50:21
2023-12-04 08:50:21.649 ERROR --- o.s.boot.SpringApplication               : Application run failed
04.12.2023
08:50:21
04.12.2023
08:50:21
org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'liquibase' defined in class path resource [org/springframework/boot/autoconfigure/liquibase/LiquibaseAutoConfiguration$LiquibaseConfiguration.class]: Invocation of init method failed; nested exception is liquibase.exception.CommandExecutionException: liquibase.exception.LiquibaseException: liquibase.exception.MigrationFailedException: Migration failed for changeset liquibase/11.1/add-genre-pk.xml::add_genre_pk::kagemomiji:
04.12.2023
08:50:21
     Reason: liquibase.exception.DatabaseException: (conn=254) Duplicate entry 'AudioBook' for key 'PRIMARY' [Failed SQL: (1062) ALTER TABLE `airsonic-advanced`.genre ADD PRIMARY KEY (name)]
04.12.2023
08:50:21
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1804) ~[spring-beans-5.3.30.jar:5.3.30]
04.12.2023
08:50:21
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:620) ~[spring-beans-5.3.30.jar:5.3.30]
04.12.2023
08:50:21
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:542) ~[spring-beans-5.3.30.jar:5.3.30]
04.12.2023
08:50:21
	at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:335) ~[spring-beans-5.3.30.jar:5.3.30]
04.12.2023
08:50:21
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:234) ~[spring-beans-5.3.30.jar:5.3.30]
04.12.2023
08:50:21
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:333) ~[spring-beans-5.3.30.jar:5.3.30]
04.12.2023
08:50:21
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:208) ~[spring-beans-5.3.30.jar:5.3.30]
04.12.2023
08:50:21
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:322) ~[spring-beans-5.3.30.jar:5.3.30]
04.12.2023
08:50:21
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:208) ~[spring-beans-5.3.30.jar:5.3.30]
04.12.2023
08:50:21
	at org.springframework.context.support.AbstractApplicationContext.getBean(AbstractApplicationContext.java:1160) ~[spring-context-5.3.30.jar:5.3.30]
04.12.2023
08:50:21
	at org.springframework.context.support.AbstractApplicationContext.finishBeanFactoryInitialization(AbstractApplicationContext.java:911) ~[spring-context-5.3.30.jar:5.3.30]
04.12.2023
08:50:21
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:583) ~[spring-context-5.3.30.jar:5.3.30]
04.12.2023
08:50:21
	at org.springframework.boot.web.servlet.context.ServletWebServerApplicationContext.refresh(ServletWebServerApplicationContext.java:147) ~[spring-boot-2.7.16.jar:2.7.16]
04.12.2023
08:50:21
	at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:731) ~[spring-boot-2.7.16.jar:2.7.16]
04.12.2023
08:50:21
	at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:408) ~[spring-boot-2.7.16.jar:2.7.16]
04.12.2023
08:50:21
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:307) ~[spring-boot-2.7.16.jar:2.7.16]
04.12.2023
08:50:21
	at org.springframework.boot.builder.SpringApplicationBuilder.run(SpringApplicationBuilder.java:164) ~[spring-boot-2.7.16.jar:2.7.16]
04.12.2023
08:50:21
	at org.airsonic.player.Application.main(Application.java:70) ~[classes/:na]
04.12.2023
08:50:21
Caused by: liquibase.exception.CommandExecutionException: liquibase.exception.LiquibaseException: liquibase.exception.MigrationFailedException: Migration failed for changeset liquibase/11.1/add-genre-pk.xml::add_genre_pk::kagemomiji:
04.12.2023
08:50:21
     Reason: liquibase.exception.DatabaseException: (conn=254) Duplicate entry 'AudioBook' for key 'PRIMARY' [Failed SQL: (1062) ALTER TABLE `airsonic-advanced`.genre ADD PRIMARY KEY (name)]
04.12.2023
08:50:21
	at liquibase.command.CommandScope.execute(CommandScope.java:235) ~[liquibase-core-4.22.0.jar:na]
04.12.2023
08:50:21
	at liquibase.Liquibase.lambda$update$0(Liquibase.java:237) ~[liquibase-core-4.22.0.jar:na]
04.12.2023
08:50:21
	at liquibase.Scope.lambda$child$0(Scope.java:194) ~[liquibase-core-4.22.0.jar:na]
04.12.2023
08:50:21
	at liquibase.Scope.child(Scope.java:203) ~[liquibase-core-4.22.0.jar:na]
04.12.2023
08:50:21
	at liquibase.Scope.child(Scope.java:193) ~[liquibase-core-4.22.0.jar:na]
04.12.2023
08:50:21
	at liquibase.Scope.child(Scope.java:172) ~[liquibase-core-4.22.0.jar:na]
04.12.2023
08:50:21
	at liquibase.Liquibase.runInScope(Liquibase.java:1605) ~[liquibase-core-4.22.0.jar:na]
04.12.2023
08:50:21
	at liquibase.Liquibase.update(Liquibase.java:229) ~[liquibase-core-4.22.0.jar:na]
04.12.2023
08:50:21
	at liquibase.Liquibase.update(Liquibase.java:213) ~[liquibase-core-4.22.0.jar:na]
04.12.2023
08:50:21
	at liquibase.integration.spring.SpringLiquibase.performUpdate(SpringLiquibase.java:329) ~[liquibase-core-4.22.0.jar:na]
04.12.2023
08:50:21
	at liquibase.integration.spring.SpringLiquibase.afterPropertiesSet(SpringLiquibase.java:284) ~[liquibase-core-4.22.0.jar:na]
04.12.2023
08:50:21
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.invokeInitMethods(AbstractAutowireCapableBeanFactory.java:1863) ~[spring-beans-5.3.30.jar:5.3.30]
04.12.2023
08:50:21
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1800) ~[spring-beans-5.3.30.jar:5.3.30]
04.12.2023
08:50:21
	... 17 common frames omitted
04.12.2023
08:50:21
Caused by: liquibase.exception.LiquibaseException: liquibase.exception.MigrationFailedException: Migration failed for changeset liquibase/11.1/add-genre-pk.xml::add_genre_pk::kagemomiji:
04.12.2023
08:50:21
     Reason: liquibase.exception.DatabaseException: (conn=254) Duplicate entry 'AudioBook' for key 'PRIMARY' [Failed SQL: (1062) ALTER TABLE `airsonic-advanced`.genre ADD PRIMARY KEY (name)]
04.12.2023
08:50:21
	at liquibase.changelog.ChangeLogIterator.run(ChangeLogIterator.java:121) ~[liquibase-core-4.22.0.jar:na]
04.12.2023
08:50:21
	at liquibase.command.core.AbstractUpdateCommandStep.lambda$run$0(AbstractUpdateCommandStep.java:103) ~[liquibase-core-4.22.0.jar:na]
04.12.2023
08:50:21
	at liquibase.Scope.lambda$child$0(Scope.java:194) ~[liquibase-core-4.22.0.jar:na]
04.12.2023
08:50:21
	at liquibase.Scope.child(Scope.java:203) ~[liquibase-core-4.22.0.jar:na]
04.12.2023
08:50:21
	at liquibase.Scope.child(Scope.java:193) ~[liquibase-core-4.22.0.jar:na]
04.12.2023
08:50:21
	at liquibase.Scope.child(Scope.java:172) ~[liquibase-core-4.22.0.jar:na]
04.12.2023
08:50:21
	at liquibase.command.core.AbstractUpdateCommandStep.run(AbstractUpdateCommandStep.java:100) ~[liquibase-core-4.22.0.jar:na]
04.12.2023
08:50:21
	at liquibase.command.CommandScope.execute(CommandScope.java:212) ~[liquibase-core-4.22.0.jar:na]
04.12.2023
08:50:21
	... 29 common frames omitted
04.12.2023
08:50:21
Caused by: liquibase.exception.MigrationFailedException: Migration failed for changeset liquibase/11.1/add-genre-pk.xml::add_genre_pk::kagemomiji:
04.12.2023
08:50:21
     Reason: liquibase.exception.DatabaseException: (conn=254) Duplicate entry 'AudioBook' for key 'PRIMARY' [Failed SQL: (1062) ALTER TABLE `airsonic-advanced`.genre ADD PRIMARY KEY (name)]
04.12.2023
08:50:21
	at liquibase.changelog.ChangeSet.execute(ChangeSet.java:759) ~[liquibase-core-4.22.0.jar:na]
04.12.2023
08:50:21
	at liquibase.changelog.visitor.UpdateVisitor.visit(UpdateVisitor.java:61) ~[liquibase-core-4.22.0.jar:na]
04.12.2023
08:50:21
	at liquibase.changelog.ChangeLogIterator$2.lambda$null$0(ChangeLogIterator.java:109) ~[liquibase-core-4.22.0.jar:na]
04.12.2023
08:50:21
	at liquibase.Scope.lambda$child$0(Scope.java:194) ~[liquibase-core-4.22.0.jar:na]
04.12.2023
08:50:21
	at liquibase.Scope.child(Scope.java:203) ~[liquibase-core-4.22.0.jar:na]
04.12.2023
08:50:21
	at liquibase.Scope.child(Scope.java:193) ~[liquibase-core-4.22.0.jar:na]
04.12.2023
08:50:21
	at liquibase.Scope.child(Scope.java:172) ~[liquibase-core-4.22.0.jar:na]
04.12.2023
08:50:21
	at liquibase.changelog.ChangeLogIterator$2.lambda$run$1(ChangeLogIterator.java:109) ~[liquibase-core-4.22.0.jar:na]
04.12.2023
08:50:21
	at liquibase.Scope.lambda$child$0(Scope.java:194) ~[liquibase-core-4.22.0.jar:na]
04.12.2023
08:50:21
	at liquibase.Scope.child(Scope.java:203) ~[liquibase-core-4.22.0.jar:na]
04.12.2023
08:50:21
	at liquibase.Scope.child(Scope.java:193) ~[liquibase-core-4.22.0.jar:na]
04.12.2023
08:50:21
	at liquibase.Scope.child(Scope.java:172) ~[liquibase-core-4.22.0.jar:na]
04.12.2023
08:50:21
	at liquibase.Scope.child(Scope.java:260) ~[liquibase-core-4.22.0.jar:na]
04.12.2023
08:50:21
	at liquibase.changelog.ChangeLogIterator$2.run(ChangeLogIterator.java:92) ~[liquibase-core-4.22.0.jar:na]
04.12.2023
08:50:21
	at liquibase.Scope.lambda$child$0(Scope.java:194) ~[liquibase-core-4.22.0.jar:na]
04.12.2023
08:50:21
	at liquibase.Scope.child(Scope.java:203) ~[liquibase-core-4.22.0.jar:na]
04.12.2023
08:50:21
	at liquibase.Scope.child(Scope.java:193) ~[liquibase-core-4.22.0.jar:na]
04.12.2023
08:50:21
	at liquibase.Scope.child(Scope.java:172) ~[liquibase-core-4.22.0.jar:na]
04.12.2023
08:50:21
	at liquibase.Scope.child(Scope.java:260) ~[liquibase-core-4.22.0.jar:na]
04.12.2023
08:50:21
	at liquibase.Scope.child(Scope.java:264) ~[liquibase-core-4.22.0.jar:na]
04.12.2023
08:50:21
	at liquibase.changelog.ChangeLogIterator.run(ChangeLogIterator.java:66) ~[liquibase-core-4.22.0.jar:na]
04.12.2023
08:50:21
	... 36 common frames omitted
04.12.2023
08:50:21
Caused by: liquibase.exception.DatabaseException: (conn=254) Duplicate entry 'AudioBook' for key 'PRIMARY' [Failed SQL: (1062) ALTER TABLE `airsonic-advanced`.genre ADD PRIMARY KEY (name)]
04.12.2023
08:50:21
	at liquibase.executor.jvm.JdbcExecutor$ExecuteStatementCallback.doInStatement(JdbcExecutor.java:445) ~[liquibase-core-4.22.0.jar:na]
04.12.2023
08:50:21
	at liquibase.executor.jvm.JdbcExecutor.execute(JdbcExecutor.java:77) ~[liquibase-core-4.22.0.jar:na]
04.12.2023
08:50:21
	at liquibase.executor.jvm.JdbcExecutor.execute(JdbcExecutor.java:160) ~[liquibase-core-4.22.0.jar:na]
04.12.2023
08:50:21
	at liquibase.database.AbstractJdbcDatabase.execute(AbstractJdbcDatabase.java:1287) ~[liquibase-core-4.22.0.jar:na]
04.12.2023
08:50:21
	at liquibase.database.AbstractJdbcDatabase.executeStatements(AbstractJdbcDatabase.java:1269) ~[liquibase-core-4.22.0.jar:na]
04.12.2023
08:50:21
	at liquibase.changelog.ChangeSet.execute(ChangeSet.java:717) ~[liquibase-core-4.22.0.jar:na]
04.12.2023
08:50:21
	... 56 common frames omitted
04.12.2023
08:50:21
Caused by: java.sql.SQLIntegrityConstraintViolationException: (conn=254) Duplicate entry 'AudioBook' for key 'PRIMARY'
04.12.2023
08:50:21
	at org.mariadb.jdbc.export.ExceptionFactory.createException(ExceptionFactory.java:290) ~[mariadb-java-client-3.1.4.jar:na]
04.12.2023
08:50:21
	at org.mariadb.jdbc.export.ExceptionFactory.create(ExceptionFactory.java:370) ~[mariadb-java-client-3.1.4.jar:na]
04.12.2023
08:50:21
	at org.mariadb.jdbc.message.ClientMessage.readPacket(ClientMessage.java:134) ~[mariadb-java-client-3.1.4.jar:na]
04.12.2023
08:50:21
	at org.mariadb.jdbc.client.impl.StandardClient.readPacket(StandardClient.java:872) ~[mariadb-java-client-3.1.4.jar:na]
04.12.2023
08:50:21
	at org.mariadb.jdbc.client.impl.StandardClient.readResults(StandardClient.java:811) ~[mariadb-java-client-3.1.4.jar:na]
04.12.2023
08:50:21
	at org.mariadb.jdbc.client.impl.StandardClient.readResponse(StandardClient.java:730) ~[mariadb-java-client-3.1.4.jar:na]
04.12.2023
08:50:21
	at org.mariadb.jdbc.client.impl.StandardClient.execute(StandardClient.java:654) ~[mariadb-java-client-3.1.4.jar:na]
04.12.2023
08:50:21
	at org.mariadb.jdbc.Statement.executeInternal(Statement.java:957) ~[mariadb-java-client-3.1.4.jar:na]
04.12.2023
08:50:21
	at org.mariadb.jdbc.Statement.execute(Statement.java:1083) ~[mariadb-java-client-3.1.4.jar:na]
04.12.2023
08:50:21
	at org.mariadb.jdbc.Statement.execute(Statement.java:474) ~[mariadb-java-client-3.1.4.jar:na]
04.12.2023
08:50:21
	at com.zaxxer.hikari.pool.ProxyStatement.execute(ProxyStatement.java:94) ~[HikariCP-4.0.3.jar:na]
04.12.2023
08:50:21
	at com.zaxxer.hikari.pool.HikariProxyStatement.execute(HikariProxyStatement.java) ~[HikariCP-4.0.3.jar:na]
04.12.2023
08:50:21
	at liquibase.executor.jvm.JdbcExecutor$ExecuteStatementCallback.doInStatement(JdbcExecutor.java:441) ~[liquibase-core-4.22.0.jar:na]
04.12.2023
08:50:21
	... 61 common frames omitted

--- a/airsonic-main/src/main/resources/liquibase/11.1/changelog.xml
+++ b/airsonic-main/src/main/resources/liquibase/11.1/changelog.xml
@@ -4,5 +4,6 @@
         xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.1.xsd">
     <include file="add-user-credentials-id.xml" relativeToChangelogFile="true"/>
     <include file="add-deleted-column-music-folder.xml" relativeToChangelogFile="true"/>
+    <include file="remove-genre-name-duplication.xml" relativeToChangelogFile="true"/>
     <include file="add-genre-pk.xml" relativeToChangelogFile="true"/>
 </databaseChangeLog>

--- a/airsonic-main/src/main/resources/liquibase/11.1/remove-genre-name-duplication.xml
+++ b/airsonic-main/src/main/resources/liquibase/11.1/remove-genre-name-duplication.xml
@@ -1,0 +1,59 @@
+<databaseChangeLog xmlns="http://www.liquibase.org/xml/ns/dbchangelog"
+                   xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+                   xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.1.xsd">
+    <!-- This change set is to remove the duplication of genre name in the database. -->
+    <!-- rename genre to genre_backup-->
+    <changeSet id="rename-genre-to-backup" author="kagemomiji">
+        <preConditions onFail="MARK_RAN">
+            <not>
+                <primaryKeyExists tableName="genre"/>
+            </not>
+        </preConditions>
+        <renameTable newTableName="genre_backup" oldTableName="genre"/>
+    </changeSet>
+
+    <!-- crete new genre table -->
+    <changeSet id="create-new-genre-table" author="kagemomiji">
+        <preConditions onFail="MARK_RAN">
+            <not>
+                <primaryKeyExists tableName="genre"/>
+            </not>
+        </preConditions>
+        <createTable tableName="genre">
+            <column name="name" type="${varchar_type}"/>
+            <column name="song_count" type="integer"/>
+            <column name="album_count" type="integer"/>
+        </createTable>
+    </changeSet>
+
+    <!-- Copy data from genre_backup to genre -->
+    <changeSet id="copy-distinct-genre-data" author="kagemomiji">
+        <preConditions onFail="MARK_RAN">
+            <not>
+                <primaryKeyExists tableName="genre"/>
+            </not>
+        </preConditions>
+        <sql>
+            INSERT INTO genre (name, song_count, album_count)
+            SELECT DISTINCT name, song_count, album_count FROM genre_backup;
+        </sql>
+        <rollback>
+            <dropTable tableName="genre"/>
+            <renameTable oldTableName="genre_backup" newTableName="genre"/>
+        </rollback>
+    </changeSet>
+
+    <!-- Drop the genre_backup table -->
+    <changeSet id="drop-genre-backup-table" author="kagemomiji">
+        <preConditions onFail="MARK_RAN">
+            <not>
+                <primaryKeyExists tableName="genre"/>
+            </not>
+        </preConditions>
+        <dropTable tableName="genre_backup"/>
+        <rollback>
+            <dropTable tableName="genre"/>
+            <renameTable oldTableName="genre_backup" newTableName="genre"/>
+        </rollback>
+    </changeSet>
+</databaseChangeLog>

--- a/airsonic-main/src/main/resources/liquibase/11.1/remove-genre-name-duplication.xml
+++ b/airsonic-main/src/main/resources/liquibase/11.1/remove-genre-name-duplication.xml
@@ -35,7 +35,9 @@
         </preConditions>
         <sql>
             INSERT INTO genre (name, song_count, album_count)
-            SELECT DISTINCT name, song_count, album_count FROM genre_backup;
+            SELECT name, MAX(song_count), MAX(album_count)
+            FROM genre_backup
+            GROUP BY name;
         </sql>
         <rollback>
             <dropTable tableName="genre"/>

